lib/ProkkaAnnotation/Util/ProkkaUtils.py

```diff
@@ -0,0 +1,699 @@
+                    genome_data["data"]["features"][i]["functions"] = [new_function]
+                    stats['found_functions'] += 1
+
+                # Set Ontologies
+                new_ontology = new_annotations[fid].get("ontology_terms", None)
+                if new_ontology:
+                    stats['new_ontologies'] += 1
+                    if new_genome:
+                        genome_data["data"]["features"][i] = self. \
+                            new_genome_ontologies(feature, new_ontology, ontology_event_index)
+                    else:
+                        genome_data["data"]["features"][i] = self. \
+                            old_genome_ontologies(feature, new_ontology)
+            if current_function:
+                func_r.write(json.dumps([fid, [current_function], [new_function]]) + "\n")
+            else:
+                func_r.write(json.dumps([fid, current_functions, [new_function]]) + "\n")
+
+            onto_r.write(json.dumps([fid, current_ontology, new_ontology]) + "\n")
+
+        func_r.close()
+        onto_r.close()
+
+        info = self.gfu.save_one_genome({"workspace": self.output_workspace,
+                                         "name": annotation_args["output_genome_name"],
+                                         "data": genome_data["data"],
+                                         "provenance": self.ctx.provenance()})["info"]
+
+        genome_ref = str(info[6]) + "/" + str(info[0]) + "/" + str(info[4])
+
+        annotated_genome = namedtuple("annotated_genome",
+                                      "genome_ref function_report_filepath ontology_report_filepath stats")
+
+        return annotated_genome(genome_ref, function_report_filepath, ontology_report_filepath,
+                                stats)
+
+    def upload_file(self, filepath, message="Annotation report generated by kb_prokka"):
+        """
+        Upload a file to shock
+        :param filepath: File to upload
+        :param message: Optional Upload Message
+        :return:
+        """
+        output_file_shock_id = self.dfu.file_to_shock({"file_path": filepath})["shock_id"]
+        print("Uploaded filepath" + filepath + "to shock and got id" + output_file_shock_id)
+        return {"shock_id": output_file_shock_id,
+                "name": os.path.basename(filepath),
+                "label": os.path.basename(filepath),
+                "description": message}
+
+    def report_annotated_genome(self, genome):
+        """ Create report output with newly reannotated genome, and some stats
+
+        :param genome: Reannotated Genome Reference, Report Files and Stats
+        :return: Reference to Report Object
+        """
+        genome_ref = genome.genome_ref
+        stats = genome.stats
+
+        file_links = [self.upload_file(genome.ontology_report_filepath),
+                      self.upload_file(genome.function_report_filepath)]
```

Comment:
Best not to call either of these "reports" because that usually means something different in a KBase context. Maybe "summaries"?

Reply:
ok done